mayastor-test/test_rebuild.js

```diff
@@ -0,0 +1,379 @@
+    client.ListNexus({}, (err, res, done) => {
+      if (err) return done(err);
+      assert.lengthOf(res.nexus_list, 1);
+
+      let nexus = res.nexus_list[0];
+      assert.equal(nexus.uuid, UUID);
+
+      assert.equal(nexus.rebuilds, expected);
+    });
+  }
+
+  function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  const createNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.createNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const destroyNexus = args => {
+    return new Promise((resolve, reject) => {
+      client.destroyNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const addChild = args => {
+    return new Promise((resolve, reject) => {
+      client.addChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const removeChild = args => {
+    return new Promise((resolve, reject) => {
+      client.removeChildNexus(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const startRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.startRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const stopRebuild = args => {
+    return new Promise((resolve, reject) => {
+      client.stopRebuild(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  const rebuildState = args => {
+    return new Promise((resolve, reject) => {
+      client.getRebuildState(args, (err, data) => {
+        if (err) return reject(err);
+        resolve(data);
+      });
+    });
+  };
+
+  before(done => {
+    client = createGrpcClient('MayaStor');
+    if (!client) {
+      return done(new Error('Failed to initialize grpc client'));
+    }
+
+    async.series(
+      [
+        common.ensureNbdWritable,
+        next => {
+          fs.writeFile(child1, '', next);
+        },
+        next => {
+          fs.truncate(child1, diskSize, next);
+        },
+        next => {
+          fs.writeFile(child2, '', next);
+        },
+        next => {
+          fs.truncate(child2, diskSize, next);
+        },
+        next => {
+          common.startMayastor(configNexus, ['-r', common.SOCK, '-s', 386]);
+          common.startMayastorGrpc();
+          common.waitFor(pingDone => {
+            // use harmless method to test if the mayastor is up and running
+            client.listPools({}, pingDone);
+          }, next);
+        },
+        next => {
+          createNexus(nexusArgs)
+            .then(() => {
+              next();
+            })
+            .catch(err => {
+              assert(err);
+            })
+            .catch(done);
```

Comment:
IMO the three lines above this one are not needed. The last catch calls done callback with error argument.

Reply:
Done - removed unnecessary lines